cylc/flow/config.py

```diff
@@ -2349,16 +2349,16 @@ def check_for_owner(tasks: Dict) -> None:
         """
         owners = {}
         for task, tdef in tasks.items():
-            remote = tdef.get('remote', None)
-            if remote:
-                owner = remote.get('owner', None)
-                if owner:
-                    owners[task] = owner
+            owner = tdef.get('remote', {}).get('owner', None)
+            if owner:
+                owners[task] = owner
         if owners:
+            # TODO: Convert URL to a stable or latest release doc after 8.0
+            # https://github.com/cylc/cylc-flow/issues/4663
             msg = (
-                '"[runtime][task][remote]owner" is deprecated at Cylc 8.'
-                '\nsee https://cylc.github.io/cylc-doc/latest/'
-                'html/7-to-8/index.html'
+                '"[runtime][task][remote]owner" is obsolete at Cylc 8.'
+                '\nsee https://cylc.github.io/cylc-doc/nightly/'
```

Comment:
```suggestion
                '\nsee https://cylc.github.io/cylc-doc/latest/'
```

Reply:
(explained in comment) won't work because the page isn't live yet